src/components/Dataset/MoveToDatasetModal.js

```diff
@@ -10,49 +10,53 @@ import { Row } from 'components/shared/Row'
 
 export const MoveToDatasetModal = ({
   id,
+  closeModal,
   defaultValue,
   dataset,
+  pathname,
   radioOptions,
   value,
-  closeModal,
   setValue
 }) => {
-  const { addSamples, getTotalSamples, replaceSamples } = useDatasetManager()
+  const {
+    dataset: datasetState,
+    loading,
+    addSamples,
+    getTotalSamples,
+    replaceSamples
+  } = useDatasetManager()
   const { setResponsive } = useResponsive()
+  const { push } = useRouter()
+  const totalSamples = formatNumbers(getTotalSamples(datasetState.data))
+  const newDatasetTotalSamples = formatNumbers(getTotalSamples(dataset.data))
 
-  const router = useRouter()
-  const moveToDataSet = async (action = 'append') => {
+  const handleMoveSamples = async (action = 'append') => {
     if (action === 'append') {
       await addSamples(dataset.data)
-      router.push(
+      push(
         {
-          pathname: '/download',
+          pathname,
           query: {
-            message: `Appended ${formatNumbers(
-              getTotalSamples(dataset.data)
-            )} samples to My Dataset`,
+            message: `Appended ${newDatasetTotalSamples} samples to My Dataset`,
```

Comment:
We should avoid passing in formatted strings as query parameters. We should instead pass the data for the component to use as needed.
 
 This applies to line 52 below as well. 

```suggestion
            samplesCount: newDatasetTotalSamples,
            action: "Appended",
```

Reply:
The parameter `message` (its value as template literals) is used in the shared [`Notification`](https://github.com/AlexsLemonade/refinebio-web/blob/dev/src/components/shared/Notification.js) component which allows this shared component to be flexible and generic (takes any messages, not specific data values as parameters).

The router's `query` object in our application is "watched" by `Notification` from `Layout` (at the top level). When the `message` and `status` are added to the `query` object, that `message` value will be rendered in UI using the matching theme for the `status` (for example, `success` is green, `error` is coral). 

If we only pass the data value instead of the whole message value, we would have to build a new custom component in order to render the exact message that goes with that data value. 

For instance, for ths move to dataset modal, we'll need to pass the data value for `newSampleCounts`, `action`, and `status` etc, and if we need to display another notification for a particular feature, each of these also needs its own special custom component and specific parameters (and that also makes our codebase less DRY).

Could we please keep this `message` parameter? Or please let me know if there is a better alternative 💡

Reply:
I see what you mean.

Placing that value in local storage via a context would be more consistent with how we manage session variables.

I think this is ok for now and moving that over can be done in another ticket.

We need to add a default option for `status` in the `Notification` component or null check because it crashes if only `message` is in the query params as it is right now.

Please add that in this PR and then this is good for now.